reconcile/ocm/utils.py

```diff
@@ -1,10 +1,32 @@
 from collections.abc import Mapping
-from typing import Any
+from typing import (
+    Any,
+    Optional,
+    Protocol,
+    Union,
+    runtime_checkable,
+)
 
 
-def cluster_disabled_integrations(cluster: Mapping[str, Any]) -> list[str]:
-    disabled_ints: list[str] = []
-    disable = cluster.get("disable")
-    if disable:
-        disabled_ints = disable.get("integrations") or []
-    return disabled_ints
+class ClusterDisableIntegration(Protocol):
+    integrations: Optional[list[str]]
+
+
+@runtime_checkable
+class Cluster(Protocol):
+    @property
+    def disable(self) -> Optional[ClusterDisableIntegration]:
+        pass
```

Comment:
there is an opportunity here to not make this about clusters but about disabling integrations in a specific context. see the `/aws/account-1.yml`schema. it has the same `disable.integrations` config options as `/openshift/cluster-1.yml`

proposal: rename this to `SupportsIntegrationDisable` and move it to a general utils file outside of the ocm module.

feel free to ignore this in this PR, but i know that you are in a refactoring frenzy right now, so maybe something you would like to tackle in upcoming PRs :D

Reply:
Sounds like a good idea. Let me do that right now.

Reply:
I've consolidated all `disable.integrations` for clusters and accounts. Please have a look

Reply:
nice. 

i hate to be a PITA about naming, especially since i was bugging you with the `Supports*` vs not argument on a previous PR, but in this case i think `SupportsIntegrationDisable` or something along that lines would be the fitting name because the protocol describes a supported behaviour, namely "disabling integrations in a context".

i'm not going to die on this hill and annoy you with this, so again feel free to ignore 💋

Reply:
![image](https://user-images.githubusercontent.com/366430/205086441-34ccf36b-8861-4653-beab-6c964f575b19.png)

Reply:
I won't on die on that hill; naming things is still the hardest problem in IT :) I was referring to
https://github.com/python/typeshed/pull/4229/files and https://gitlab.cee.redhat.com/service/dev-guidelines/-/merge_requests/144

> "Use SupportX for protocols that provide callable methods ...". 

What do you think, should it be `HasDisable`?

Reply:
a bit generic of a name especially since it contains an `integrations` property

how about `HasXXXXIntegrationXXX`, e.g. `HasIntegrationDisableToggle`?

Reply:
What about this?

```python
class HasIntegrations():
    integrations: ...

class HasDisableIntegrations():
    disable: HasIntegrations
```

Reply:
Sounds good. 👍